seven_half/game.py

```diff
@@ -105,4 +108,44 @@ def check_card(self) -> Card:
 
 
 class Table:
-    ...
+    """Represent the table where the game happens."""
+
+    def __init__(self, deck: Deck):
+        """Instantiate a new table."""
+        self._players: list[Player] = []
+        self._score: dict[Player, int] = {}
+        self._deck: Deck = deck
+
+    def get_players(self) -> list[Player]:
+        """returns the list of players"""
+        return self._players
+
+    def add_players(self, *players) -> None:
+        """add a player or multiple players"""
+        for player in players:
+            self._players.append(player)
+
+    def remove_players(self, *players) -> None:
+        """remove a player or multiple players"""
+        for player in players:
+            self._players.remove(player)
+
+    def get_score(self) -> dict[Player, int]:
+        """returns the score as a dictionary"""
+        return self._score
+
+    def increase_score(self, player: Player, add: int) -> None:
+        """increase the score of a player by a certain amount"""
+        self._score[player] += add
+
+    def reset_score(self) -> None:
+        """reset the score to zero"""
+        self._score = 0
+
+    def get_deck(self) -> Deck:
+        """returns the deck"""
+        return self._deck
+
+    def get_cards_from_deck(self) -> Card:
+        """return the next card in the deck"""
+        yield from self._deck
```

Comment:
- Revert all implementations to `...`, type hint methods as shown in the "Class structure" specs sheet, as you were assigned to add the methods and their corresponding docstrings, not implement them.
- Refer to `Player` class' methods' docstrings on style, tone and information conveyed. Example: 
```py
    def quit(self) -> None:
        """
        Quit the round.

        This method sets the Player object's lost attribute to True.
        """
```

```py
    def change_4_card(self) -> None:
        """
        Change the 4-card in player's hand for a new one.

        This behaviour is only valid for the first card dealt. If this method is called after the
        first card, it is ignored.
        """
```